applications/FluidDynamicsApplication/custom_elements/vms_adjoint_element.h

```diff
@@ -461,13 +567,57 @@ class KRATOS_API(FLUID_DYNAMICS_APPLICATION) VMSAdjointElement: public Element
 
     void CalculateMassMatrix(
         MatrixType& rMassMatrix,
-        const ProcessInfo& rProcessInfo) override
+        const ProcessInfo& rCurrentProcessInfo) override
     {
-        KRATOS_TRY
+        // Resize and set to zero
+        if (rMassMatrix.size1() != TFluidLocalSize)
+            rMassMatrix.resize(TFluidLocalSize, TFluidLocalSize, false);
 
-        KRATOS_ERROR << "this function is not implemented.";
+        rMassMatrix = ZeroMatrix(TFluidLocalSize, TFluidLocalSize);
 
-        KRATOS_CATCH("")
+        const auto& r_geometry = this->GetGeometry();
+
+        // Get the element's geometric parameters
+        double Area;
+        array_1d<double, TNumNodes> N;
+        BoundedMatrix<double, TNumNodes, TDim> DN_DX;
+        GeometryUtils::CalculateGeometryData(r_geometry, DN_DX, N, Area);
+
+        double density, viscosity;
+        array_1d<double, TDim> velocity, mesh_velocity;
+        FluidCalculationUtilities::EvaluateInPoint(
+            r_geometry, N,
+            std::tie(density, DENSITY),
+            std::tie(velocity, VELOCITY),
+            std::tie(mesh_velocity, MESH_VELOCITY),
+            std::tie(viscosity, VISCOSITY));
+
+        viscosity *= density;
+
+        // Add 'classical' mass matrix (lumped)
+        double Coeff = density * Area / TNumNodes; //Optimize!
+        unsigned int DofIndex = 0;
+        for (unsigned int iNode = 0; iNode < TNumNodes; ++iNode)
+        {
+            for (unsigned int d = 0; d < TDim; ++d)
+            {
+                rMassMatrix(DofIndex, DofIndex) += Coeff;
+                ++DofIndex;
+            }
+            ++DofIndex; // Skip pressure Dof
+        }
+
+        // Get Advective velocity
+        const array_1d<double, TDim> effective_velocity = velocity - mesh_velocity;
+
+        // stabilization parameters
+        const double ElemSize = this->CalculateElementSize(Area);
```

Comment:
Same comment.

Reply:
It is to be consistent with the `VMS` element. That's the only reason.

Reply:
Same.